src/raftstore/store/fsm/peer.rs

```diff
@@ -1859,12 +1857,23 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
                 &mut self.fsm.peer,
             );
         }
+        let target = state.get_target().get_id();
         self.fsm.peer.pending_merge_state = Some(state);
         self.notify_prepare_merge();
 
-        if merged {
-            // CommitMerge will try to catch up log for source region. If PrepareMerge is executed
-            // in the progress of catching up, there is no need to schedule merge again.
+        if let Some(logs_up_to_date) = self.fsm.peer.catch_up_logs.take() {
+            logs_up_to_date.store(region.get_id(), Ordering::SeqCst);
```

Comment:
why not make it as a `boolean`?

Reply:
see Line 1796 in `apply.rs`, it's just a double check.